developer/images/dependencies-update/content/bin/tasks/update_binaries.sh

```diff
@@ -119,37 +103,35 @@ get_oc_version() {
 }
 
 get_shellcheck_version() {
-    get_github_release "https://github.com/koalaman/shellcheck/releases/latest"
+    get_github_release "https://github.com/koalaman/shellcheck"
 }
 
 get_tektoncd_cli_version() {
-    get_github_release "https://github.com/tektoncd/cli/releases/latest"
+    get_github_release "https://github.com/tektoncd/cli"
 }
 
 get_yamllint_version() {
-    URL="https://github.com/adrienverge/yamllint.git"
-    VERSION=$(
-        git ls-remote --tags "$URL" \
-            | grep -vE -- "-alpha|-beta|-pre|-rc|\^\{\}" \
-            | sed "s:^.*refs/tags/::" \
-            | sort -V \
-            | tail -1
-    )
-    VERSION=$(echo "$VERSION" | sed "s:^v::")
+    get_github_release "https://github.com/adrienverge/yamllint.git" "v"
 }
 
 get_yq_version() {
-    get_github_release "https://github.com/mikefarah/yq/releases/latest"
+    get_github_release "https://github.com/mikefarah/yq"
 }
 
 get_github_release() {
     URL="$1"
+    PREFIX="${2:-}"
     VERSION=$(
-        curl --location --output /dev/null --silent --write-out "%{url_effective}" "$URL" \
-            | sed "s:.*/::"
+        git ls-remote --tags "$URL" \
+            | grep -E "$PREFIX" \
+            | grep -E "[0-9]+\.[0-9]+" \
+            | grep -vE "[0-9]-*alpha|[0-9]-*beta|[0-9]-*pre|[0-9]-*rc|\^\{\}" \
+            | sed "s:^.*refs/tags/$PREFIX::" \
+            | sort -V \
+            | tail -1
```

Comment:
@Roming22 I am sure this would work but its gotten quite complex to read and modify in the future.
Have you considered using Github's API instead with curl? 
Something like:

```
get_github_release() {
    REPO_OWNER=$1
    REPO_NAME=$2
    CURRENT_VERSION=$3
    VERSION=$(curl --silent https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/releases | jq -r '.[].tag_name' | sort -V | tail -1)
    if [ $VERSION != $CURRENT_VERSION ]
    then
        echo $VERSION
    else
        echo "Current version is up to date"
    fi
}
```